src/xrpld/app/misc/NetworkOPs.cpp

```diff
@@ -1208,11 +1218,21 @@ NetworkOPsImp::processTransaction(
     // but I'm not 100% sure yet.
     // If so, only cost is looking up HashRouter flags.
     auto const view = m_ledgerMaster.getCurrentLedger();
-    auto const [validity, reason] = checkValidity(
-        app_.getHashRouter(),
-        *transaction->getSTransaction(),
-        view->rules(),
-        app_.config());
+
+    // This function is called by several different parts of the codebase
+    // under no circumstances will we ever accept an inner txn within a batch
+    // txn from the network.
+    auto const tx = *transaction->getSTransaction();
+    if (view->rules().enabled(featureBatch) && tx.isFlag(tfInnerBatchTxn))
+    {
+        transaction->setStatus(INVALID);
+        transaction->setResult(temINVALID_BATCH);
+        app_.getHashRouter().setFlags(transaction->getID(), SF_BAD);
+        return;
+    }
+
+    auto const [validity, reason] =
+        checkValidity(app_.getHashRouter(), tx, view->rules(), app_.config());
```

Comment:
should probably move the comments on line 1217 to stick with the original code

```suggestion
    // NOTE eahennis - I think this check is redundant,
    // but I'm not 100% sure yet.
    // If so, only cost is looking up HashRouter flags.
    auto const [validity, reason] =
        checkValidity(app_.getHashRouter(), tx, view->rules(), app_.config());
```